archunit-tooling-test/build.gradle

```diff
@@ -0,0 +1,191 @@
+    if (file.exists()) {
+        println "Deleting ${file.toURI().toString()}"
+        delete file
+    } else {
+        println "Could not find ${file.toURI().toString()}"
+    }
+}
+
+task removeFromMavenLocal {
+    group = 'other'
+    description = 'Remove artifact and all versions/classifiers/flavors from local maven repository.'
+    doLast {
+        [project(':archunit-junit5-engine-api'),
+         project(':archunit-junit5-engine'),
+         project(':archunit-junit5-api'),
+         project(':archunit-junit5'),
+         project(':archunit-junit4'),
+         project(':archunit')].each { project ->
+            File file = resolveAgainstMavenLocal(getMavenRepoPath(project))
+            deleteSafely file
+        }
+
+        ['org/apache/maven/surefire', 'org/apache/maven/plugins/maven-surefire-plugin'].each { path ->
+            File file = resolveAgainstMavenLocal(path)
+            deleteSafely file
+        }
+    }
+}
+
+task publishJUnitDepsToMavenLocal {
+    group = 'other'
+    description = 'Publishes ArchUnit JUnit artifacts to local Maven repo'
+    dependsOn ':archunit-junit5-engine-api:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit-junit5-engine:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit-junit5-api:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit-junit5:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit-junit4:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit:publishMavenJavaPublicationToMavenLocal'
+}
+
+task chekoutCustomSurefireVersion {
+    group = 'other'
+    description = 'Checks out a custom version of Maven Surefire'
+    onlyIf { task ->
+        !(new File("${buildDir}/custom-github-deps/maven-surefire/pom.xml")).exists()
+    }
+    doLast {
+        grgit.clone(
+                dir: "${buildDir}/custom-github-deps/maven-surefire",
+                uri: 'https://github.com/crizzis/maven-surefire.git',
+                refToCheckout: 'maven-version')
+        copy {
+            into "${buildDir}/custom-github-deps/maven-surefire"
+            from(new File(project.sourceSets.testFixtures.resources.srcDirs[0], "project")) {
+                include '.mvn/wrapper/**'
+                include 'mvnw'
+            }
+        }
+    }
+}
+
+task publishCustomSurefireVersionToMavenLocal(type: Exec) {
+    group = 'other'
+    dependsOn chekoutCustomSurefireVersion
+    onlyIf { task ->
+        !resolveAgainstMavenLocal('org/apache/maven/surefire/surefire/3.0.0-custom').exists()
+    }
+    workingDir "${buildDir}/custom-github-deps/maven-surefire"
+    commandLine "./mvnw", "install", "-DskipTests"
+}
+
+task downloadCustomGradleVersion(type: Download) {
+    onlyIf { task ->
+        !(new File(project.sourceSets.main.output.resourcesDir, 'gradle-7.5-bin.zip').exists())
+    }
+    src 'https://filesend.io/8PdZ/gradle-7.5-bin.zip?download_token=6b1236a8b17cefff61063a4d4822f5b3c540a3a0c0d815eb6edcf799c83b11a1'
```

Comment:
Sorry about this one, I had to work around GitHub's 100MB file limit, and LFS doesn't work with public forks...

In case the link expires, this Gradle distribution can be built from [here](https://github.com/crizzis/gradle/tree/feature/20169)